lib/guardian/post_guardian.rb

```diff
@@ -239,10 +239,17 @@ def can_see_post?(post)
     return false unless can_see_post_topic?(post)
     return false unless post.user == @user || Topic.visible_post_types(@user).include?(post.post_type)
     return true if is_moderator? || is_category_group_moderator?(post.topic.category)
-    return true if post.deleted_at.blank? || (post.deleted_by_id == @user.id && @user.has_trust_level?(TrustLevel[4]))
+    return true if !post.trashed? || can_see_deleted_post?(post)
```

Comment:
I assume the naming mixup between "trashed" and "deleted" is inherited. 😅

Reply:
Yeah it's kind of tricky...sometimes we refer to it one way sometimes the other. The whole concept comes from the `Trashable` class...in this case there is also a `can_see_deleted_posts?` method in the guardian so I went with consistency